src/LetsTrace.Jaeger/Serialization/JaegerThriftSerialization.cs

```diff
@@ -84,10 +84,50 @@ public static List<JaegerTag> ConvertLogToJaegerTags(LogRecord log)
 
             foreach (var field in log.Fields)
             {
-                field.Marshal(tags);
+                AddToJaegerTagList(field.Key, field.Value, tags);
             }
 
             return tags;
         }
+
+        public static void AddToJaegerTagList(string key, object value, List<JaegerTag> tags)
+        {
+            switch(value)
+            {
+                case byte[] b:
+                    tags.Add(new JaegerTag{ Key = key, VType = JaegerTagType.BINARY, VBinary = b });
+                    break;
+                case string s:
+                    tags.Add(new JaegerTag{ Key = key, VType = JaegerTagType.STRING, VStr = s });
+                    break;
+                case double d:
+                    tags.Add(new JaegerTag{ Key = key, VType = JaegerTagType.DOUBLE, VDouble = d });
+                    break;
+                case decimal d:
+                    tags.Add(new JaegerTag{ Key = key, VType = JaegerTagType.DOUBLE, VDouble = (double)d });
+                    break;
+                case bool b:
+                    tags.Add(new JaegerTag{ Key = key, VType = JaegerTagType.BOOL, VBool = b });
+                    break;
+                case UInt16 u16:
```

Comment:
Nit: The case statements are now a mix between keywords (e.g. `string`) and types (e.g. `UInt32)`. You could also use `ushort`, `uint`, ... instead.

fyi: I'm not sure but I **think** you can also use the same variable for each case (e.g. `case string value:` and `case double value`). Not sure if this is more readable/better though.

Reply:
Gotcha, yeah. It would be better to be more consistent and use the keywords.